Generators/include/Generators/GeneratorHepMCParam.h

```diff
@@ -30,9 +30,9 @@ namespace eventgen
  **/
 
 struct GeneratorHepMCParam : public o2::conf::ConfigurableParamHelper<GeneratorHepMCParam> {
-  std::string fileName = "";
-  int version = 2;
+  int version = 0;
```

Comment:
You disabled this with the explanation that it can be figured out automatically from the file format.
However, for the FIFO case, a hard-coded HepMC3 reader is instantiated. 

 So either HepMC2 is not supported in this mode (which would be unfortunate) or it would make sense to keep the version field to be able to select the appropriate reader.

Reply:
Hi Sandro, 

I made a new commit to the MR, which should address your concerns (and some other improvements)

## Back-ward compatibility and Namespace change

- `GeneratorHepMC` accepts the old `HepMC.fileName` configuration
  option (with a deprecation warning)
- `HepMC.version` is honoured when spawning a child process.
  That is, if it is set to 2, then use compatibility reader
  `HepMC3::ReaderAsciiVersion2`, otherwise use the version 3
  reader.

  If option is given with file names (`GeneratorFileOrCmd.fileNames` or deprecated `HepMC3.fileName`), then give warning that it is not used, but that the reader deduces the version it self. 

- Configuration namespace `FileOrCmd` is renamed to
  `GeneratorFileOrCmd`

- Documentation has been updated to reflect these changes